src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -665,6 +665,18 @@ void SIM_RenseignementProblemeHebdo(PROBLEME_HEBDO& problem,
                 auto& inflowsmatrix = area.hydro.series->storage;
                 auto const& srcinflows = inflowsmatrix[tsIndex < inflowsmatrix.width ? tsIndex : 0];
 
+                auto& mingenmatrix = area.hydro.series->mingen;
+                if (study.header.version >= 860)
+                {
+                    auto const& srcmingen
+                      = mingenmatrix[tsIndex < mingenmatrix.width ? tsIndex : 0];
+                    for (uint j = 0; j < problem.NombreDePasDeTemps; ++j)
+                    {
+                        problem.CaracteristiquesHydrauliques[k]->MingenHoraire[j]
+                          = srcmingen[PasDeTempsDebut + j];
+                    }
+                }
```

Comment:
```suggestion
                auto& mingenmatrix = area.hydro.series->mingen;
                {
                    auto const& srcmingen
                      = mingenmatrix[tsIndex < mingenmatrix.width ? tsIndex : 0];
                    for (uint j = 0; j < problem.NombreDePasDeTemps; ++j)
                    {
                        problem.CaracteristiquesHydrauliques[k]->MingenHoraire[j]
                          = srcmingen[PasDeTempsDebut + j];
                    }
```

As much as possible, we try to avoid making distinctions based on the version for anything else than I/O

Previously, Xmin[Var] was 0. I suggest setting `mingenmatrix` to 0 coefficients by default.

Another problem with the current implementation is that if version < 860, `MingenHoraire` is unitialized, which can lead to [UB](https://en.wikipedia.org/wiki/Undefined_behavior).

@guilpier-code what's your take on this ?

Reply:
same as above